### PR TITLE
UIOR-446 Add agreement line details

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
https://issues.folio.org/browse/UIOR-446

### Screenshot
image

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/POLine/POLineAgreementLines/POLineAgreementLines.js [hunk 1]
```diff
@@ -1,22 +1,68 @@
...
+      defaultMessage={label}
+    />
+  ),
+  arrow: () => <Icon icon="caret-right" />,
+};
+
 const POLineAgreementLines = ({ agreementLines, label }) => {
   return (
     <Accordion
       displayWhenClosed={<Badge>{agreementLines.length}</Badge>}
```
Comment (aliaksei-chumakou) [inline src/components/POLine/POLineAgreementLines/POLineAgreementLines.js:52]: according to proptypes `agreementLines`could be undefined

File: test/bigtest/interactors/line-details-page.js [hunk 2]
```diff
@@ -53,4 +63,8 @@ export default interactor(class LineDetailsPage {
   whenInvoicesLoaded() {
     return this.timeout(20000).when(() => this.lineInvoicesIsPresent);
   }
+
+  whenAgreementLinesLoaded() {
+    return this.timeout(20000).when(() => this.agreementLinesArePresent);
```
Comment (aliaksei-chumakou) [inline test/bigtest/interactors/line-details-page.js:68]: probably we have a constant

File: src/components/POLine/POLineAgreementLines/POLineAgreementLines.js (hunk 1 above)
Comment (AndreiShumski2) [reply src/components/POLine/POLineAgreementLines/POLineAgreementLines.js]: agreementLines are required. Fixed it

File: src/components/POLine/POLineAgreementLines/POLineAgreementLines.js [hunk 3]
```diff
@@ -1,29 +1,82 @@
...
     >
-      {/* UIOR-446 */}
-      <div>{agreementLines.length}</div>
+      <MultiColumnList
+        columnMapping={columnMapping}
+        contentData={agreementLines}
+        formatter={resultFormatter}
+        id="po-line-agreement-lines"
+        interactive={false}
+        maxHeight={800}
```
Comment (aliaksei-chumakou) [inline src/components/POLine/POLineAgreementLines/POLineAgreementLines.js:62]: is there a problem with `autosize`?

Comment (AndreiShumski2) [reply src/components/POLine/POLineAgreementLines/POLineAgreementLines.js:62]: yes. it doesn't show MCL with `autosize` prop

File: test/bigtest/network/serializers/agreement-line.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import ApplicationSerializer from './application';
+
+export default ApplicationSerializer.extend({
+  serialize(...args) {
+    const json = ApplicationSerializer.prototype.serialize.apply(this, args);
+
+    return ({
+      results: json.agreementLines,
+      totalRecords: json.agreementLines.length,
```
Comment (aliaksei-chumakou) [inline test/bigtest/network/serializers/agreement-line.js:9]: what's wrong with `buildBaseSerializer`?

Comment (AndreiShumski2) [reply test/bigtest/network/serializers/agreement-line.js]: nothing 😊

**Pull request merged**